src/nv/storage/authorization.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    tss2_esys::{ESYS_TR, ESYS_TR_RH_OWNER, ESYS_TR_RH_PLATFORM},
+    utils::Hierarchy,
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::{From, TryFrom};
+/// Enum representing the only type of authorizations
+/// allowed.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum NvAuthorization {
+    Owner,
+    Platform,
+}
+
+impl From<NvAuthorization> for Hierarchy {
+    fn from(nv_auhtorization: NvAuthorization) -> Hierarchy {
+        match nv_auhtorization {
+            NvAuthorization::Owner => Hierarchy::Owner,
+            NvAuthorization::Platform => Hierarchy::Platform,
+        }
+    }
+}
+
+impl TryFrom<Hierarchy> for NvAuthorization {
+    type Error = Error;
+
+    fn try_from(hierarchy: Hierarchy) -> Result<NvAuthorization> {
+        match hierarchy {
+            Hierarchy::Owner => Ok(NvAuthorization::Owner),
+            Hierarchy::Platform => Ok(NvAuthorization::Platform),
+            _ => {
+                error!("Error: Found invalid value when trying to convert Hierarchy to NV Authroization.");
+                Err(Error::local_error(WrapperErrorKind::InvalidParam))
```

Comment:
Obviously not something for this PR, but I'm wondering if the concept of `InvalidParam` is waaay too dilluted in this crate. We use it for a lot of different things, maybe we should split it out into more fine-grained errors.